src/external/SMIOL/smiol.c

```diff
@@ -243,6 +244,7 @@ int SMIOL_open_file(struct SMIOL_context *context, const char *filename,
 	int io_group;
 	MPI_Comm io_file_comm;
 	MPI_Comm io_group_comm;
+	MPI_Info info = MPI_INFO_NULL;
```

Comment:
I'm not sure that there's any benefit to declaring `info` in this scope and using it in all calls to `ncmpi_open`. I'd suggest again that we declare this variable in the block beginning on line 323 (325 in this PR) and apply `info` only when creating new files.

Reply:
Since our intent is only to pass a hint to `ncmpi_create`, and not to change the semantics of the calls to `ncmpi_open` for writing an existing file or reading a file, I think it would be much cleaner and clearer if we only use `info` within the scope of the block containing the call to `ncmpi_create`.

If in future we decide to pass a non-null `info` to either of the `ncmpi_open` calls, we can decide at that time how to handle the `MPI_Info` instance.

Reply:
I think it's fine to initialize `info` at declaration.